src/build.py

```diff
@@ -693,27 +709,37 @@ def Spec():
   CopyBinaryToArchive(wasm)
 
 
-def Binaryen():
-  buildbot.Step('binaryen')
-  Mkdir(BINARYEN_OUT_DIR)
+def BinaryenBase(step_name, src_dir, out_dir, archive_dir=None):
+  buildbot.Step(step_name)
+  Mkdir(out_dir)
   proc.check_call(
-      [PREBUILT_CMAKE_BIN, '-G', 'Ninja', BINARYEN_SRC_DIR,
+      [PREBUILT_CMAKE_BIN, '-G', 'Ninja', src_dir,
        '-DCMAKE_C_COMPILER=' + CC,
        '-DCMAKE_CXX_COMPILER=' + CXX],
-      cwd=BINARYEN_OUT_DIR)
-  proc.check_call(['ninja'], cwd=BINARYEN_OUT_DIR)
-  assert os.path.isdir(BINARYEN_BIN_DIR), 'Expected %s' % BINARYEN_BIN_DIR
-  for node in os.listdir(BINARYEN_BIN_DIR):
-    f = os.path.join(BINARYEN_BIN_DIR, node)
+      cwd=out_dir)
+  proc.check_call(['ninja'], cwd=out_dir)
+  bin_dir = os.path.join(out_dir, 'bin')
+  assert os.path.isdir(bin_dir), 'Expected %s' % bin_dir
```

Comment:
Ah, I missed a crucial piece that makes it work while trying to clean up the commit history.

Basically if it's None it gets ignored when we call CopyBinaryToArchive

... I also missed the piece that actually *uses* archive_dir in the call to CopyBinaryToArchive.